src/main/java/com/mycompany/app/application/StockQuoteApplication.java

```diff
@@ -0,0 +1,47 @@
+package com.mycompany.app.application;
+
+import java.util.*;
+import java.text.SimpleDateFormat;
+import com.mycompany.app.service.*;
+import com.mycompany.app.model.*;
+
+public class StockQuoteApplication {
+
+    /**
+     * A main method which enables program execution
+     *
+     * @param args an array that should contain as elements:
+     *       {@code String} representations of a symbol, start date, and end date
+     */
+    public static void main(String[] args) {
+
+        StockServiceFactory stockServiceFactory = new StockServiceFactory();
+        StockService basicService = stockServiceFactory.createStockService("BASIC");
+		
+        StockQuote stockQuote1 = basicService.getQuote(args[0]);
```

Comment:
never access an array like this unless you are absolutely sure there is a value at the specified index. 
in this case, what would happen if this program was started with no command line args?

Reply:
ArrayIndexOutOfBoundsException